ROADMAP.md

```diff
@@ -0,0 +1,154 @@
+### Short term
+These are the core functionalities for the game engine required to build a game. Mainly focuses on using the engine as library to build upon.
+* Entity-Component System
+* Rendering
+* Input Handling
+* Audio
+* Serialization/Reflection
+* Streaming/Asset Handling
+* Documentation/Tutorial
+
+### Mid-term
+
+* Physics
+* Tooling
+* Animation
+
+### Long-term
+
+* Scripting
+* Graphical UI
+* Networking
+* AI
+
+### Develop A Game
+A game engine is built in order to build games! The best tutorial for potential users are working examples, and fancier examples are even better.
+Under this assumption we should also try to spend some time on translating our implementations into a working game, showcasing the technical aspects of our engine.
+
+## Systems/Aspects
+Discussing the scopes and features of the different subsystems in more detail.
+
+### Entity-Component System
+Our current system is based on `specs` and should probably stick to it in near future (in case some magically new library pops up..).
+As the ECS is the core part of the architecture, this needs be carefully designed and should be clear under our 3 main vision aspects:
+
+**Modularity** We expose the API from `specs` to the developers, in order to allow them to implement new components, processors and resources (larger singleton objects). The engine will have built-in processors for common operations like `Transform` to ensure spatial hierarchy.
+
+**Parallelism** Processor run in parallel with respect to task- and data-dependencies with low overhead.
+
+**Data-Driven** TODO
+
+**Next steps**
+ * Evaluating different approaches for defining task-dependencies between the processors, including support to run a processor multiple times
+ * Research ways to handle single-threaded processors (e.g nphysics) if possible
+
+### Rendering
+The trend in industry is to achieve realistic rendering, therefore most engines implement physically based rendering.
+Using `gfx-rs` for backend abstraction allows us to provide compatibility with multiple platforms without caring that much about the different graphic APIs
+
+**Modularity** TODO: easily integrate own rendering approaches
+**Parallelism** TODO: parallel command list creation
+**Data-Driven** TODO: efficient rendering -> minimizing draw calls and overhead
+
+**Next steps**
+ * Transfer to physically based rendering (pretty large tasks..)
+ * Evaluate current system in terms of extendability for developers to implement own rendering approaches
+ * Feedback practical experience into improvements for `gfx-rs`
+
+### Input Handling
+**Modularity** The API should be flexible enough to support multiple different input devices including gamepads and maybe VR devices.
+**Data-Driven** TODO: raw input
+
+**Next steps**
+ * Raw input support (winit?)
+ * Gamepad support (winit?)
+ * Input Mapping
```

Comment:
Allow the users to define custom actions like `Jump`, `Forward`, etc.
Instead of asking for `is_key_pressed(A)`, we can use the newly defined actions. Players can then remap these actions to actual input events e.g keyboard press oder gamepad events.

Will try to define this point better in the roadmap!